nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/blob/BlobToPDF.java

```diff
@@ -108,4 +142,49 @@ protected void adjustBlobName(Blob in, Blob out) {
         out.setMimeType("application/pdf");
     }
 
+    /**
+     * Replace the image URLs of an HTML blob by absolute local paths
+     *
+     * @throws IOException
+     * @since 9.1
+     */
+    @SuppressWarnings("unchecked")
+    protected Blob replaceURLsByAbsolutePaths(Blob blob, Path tempDirectory) throws IOException {
+        String initialBlobContent = blob.getString();
+        String blobContentWithAbsolutePaths = initialBlobContent;
+
+        // Check if the blob contains images to render, by parsing expressions like:
+        // src="http://localhost/nuxeo/nxfile/default/3727ef6b-cf8c-4f27-ab2c-79de0171a2c8/files:files/0/file/image.png"
+        String regex = "src=\"((.*?)/nuxeo/nxfile/(.*?)/([0-9a-z-]+)/files:files/([0-9])+/file/(.*?))\"";
```

Comment:
Ok => I added `protected getDownloadUrlRegex()` and `public parseDownloadUrlsInHtmlLinks(String content)` in `DownloadService`